consensus/state_processing/src/per_epoch_processing/epoch_processing_summary.rs

```diff
@@ -0,0 +1,280 @@
+    #[cfg(feature = "metrics")]
+    pub fn observe_metrics(&self) -> Result<(), ParticipationCacheError> {
+        metrics::set_gauge(
+            &metrics::PARTICIPATION_PREV_EPOCH_HEAD_ATTESTING_GWEI_TOTAL,
+            self.previous_epoch_head_attesting_balance()? as i64,
+        );
+        metrics::set_gauge(
+            &metrics::PARTICIPATION_PREV_EPOCH_TARGET_ATTESTING_GWEI_TOTAL,
+            self.previous_epoch_target_attesting_balance()? as i64,
+        );
+        metrics::set_gauge(
+            &metrics::PARTICIPATION_PREV_EPOCH_SOURCE_ATTESTING_GWEI_TOTAL,
+            self.previous_epoch_source_attesting_balance()? as i64,
+        );
+        metrics::set_gauge(
+            &metrics::PARTICIPATION_PREV_EPOCH_ACTIVE_GWEI_TOTAL,
+            self.previous_epoch_total_active_balance() as i64,
+        );
+
+        Ok(())
+    }
+
+    /// Returns the sum of the effective balance of all validators in the current epoch.
+    pub fn current_epoch_total_active_balance(&self) -> u64 {
+        match self {
+            EpochProcessingSummary::Base { total_balances, .. } => total_balances.current_epoch(),
+            EpochProcessingSummary::Altair {
+                participation_cache,
+            } => participation_cache.current_epoch_total_active_balance(),
+        }
+    }
+
+    /// Returns the sum of the effective balance of all validators in the current epoch who
+    /// included an attestation that matched the target.
+    pub fn current_epoch_target_attesting_balance(&self) -> Result<u64, ParticipationCacheError> {
+        match self {
+            EpochProcessingSummary::Base { total_balances, .. } => {
+                Ok(total_balances.current_epoch_target_attesters())
+            }
+            EpochProcessingSummary::Altair {
+                participation_cache,
+            } => participation_cache.current_epoch_target_attesting_balance(),
+        }
+    }
+
+    /// Returns the sum of the effective balance of all validators in the previous epoch.
+    pub fn previous_epoch_total_active_balance(&self) -> u64 {
+        match self {
+            EpochProcessingSummary::Base { total_balances, .. } => total_balances.previous_epoch(),
+            EpochProcessingSummary::Altair {
+                participation_cache,
+            } => participation_cache.previous_epoch_total_active_balance(),
+        }
+    }
+
+    /// Returns `true` if `val_index` was included in the active validator indices in the current
+    /// epoch *and* the validator is not slashed.
+    ///
+    /// ## Notes
+    ///
+    /// Always returns `false` for an unknown `val_index`.
+    pub fn is_active_unslashed_in_current_epoch(&self, val_index: usize) -> bool {
+        match self {
+            EpochProcessingSummary::Base { statuses, .. } => {
+                statuses.get(val_index).map_or(false, |s| {
+                    s.is_current_epoch_target_attester && !s.is_slashed
```

Comment:
There's a typo here:

```suggestion
                    s.is_active_in_current_epoch && !s.is_slashed
```

Reply:
Also a great catch :pray: